lib/targets/cuda/target_cuda.cmake

```diff
@@ -171,6 +171,12 @@ if(${CMAKE_CUDA_COMPILER_ID} MATCHES "NVHPC" AND NOT ${CMAKE_BUILD_TYPE} MATCHES
   target_compile_options(quda PRIVATE "$<$<COMPILE_LANG_AND_ID:CUDA,NVHPC>:SHELL: -gpu=nodebug" >)
 endif()
 
+set(QUDA_MAX_SHARED_MEMORY "0" CACHE STRING "Max shared memory per block")
```

Comment:
I'd modify the description string to say `Max shared memory per block, 0 corresponds to architecture default` or something like that. `0` gave me quite a scare when I was doing a `ccmake` configure :)